app/models/registration.rb

```diff
@@ -351,6 +351,17 @@ def self.newcomer_month_eligible_competitors_count
     errors.add(:user_id, cannot_register_reasons.to_sentence) if cannot_register_reasons.present?
   end
 
+  # TODO: This is a redunandant check with registration checker - switch to using this instead of registration_checker when
+  # functionality exists to use validations to raise errors on submitted registrations
+  validate :does_not_exceed_competitor_limit
+  private def does_not_exceed_competitor_limit
+    return if competition&.competitor_limit.blank?
+    return unless competing_status == Registrations::Helper::STATUS_ACCEPTED
+
+    errors.add(:competitor_limit, I18n.t('registrations.errors.competitor_limit_reached')) if
+      competition.registrations.accepted_and_competing_count >= competition.competitor_limit
+  end
+
```

Comment:
There is already `def cannot_exceed_competitor_limit` which does exactly the same, except:
- It only triggers when _changing_ the status to accepted (ie trying to currently accept a registration)
- It stores the error under `competing_status`, not under `competitor_limit`.

I'm quite adamant about that second point, because `competitor_limit` isn't even a field on a registration. About the first point, is that causing any troubles for you? You definitely shouldn't have to duplicate a validation like that. If anything, we should try to re-use the existing one so that it makes sense for you to use it as well

Reply:
In my defense this was added before `cannot_exceed_competitor_limit` existed